webapp/src/pages/EditChatroomDetailsPage/EditChatroomDetailsPage.tsx

```diff
@@ -0,0 +1,217 @@
+import {
+  Button,
+  ButtonVariant,
+  Header,
+  IconVariant,
+  Input,
+  InputVariant,
+  LargeAvatar,
+  Loading,
+  ToggleSwitch,
+} from '../../shared/components';
+import {
+  CHAT_URL,
+  EDIT_AVATAR_URL,
+  WILDCARD_AVATAR_URL,
+} from '../../shared/urls';
+import { useNavigate, useParams } from 'react-router-dom';
+import { useNavigation } from '../../shared/hooks/UseNavigation';
+import { useCallback, useEffect, useState } from 'react';
+import { ResponseError, UpdateChatroomDto } from '../../shared/generated';
+import './EditChatroomDetailsPage.css';
+import { chatApi } from '../../shared/services/ApiService';
+import { useData } from '../../shared/hooks/UseData';
+import { useNotificationContext } from '../../shared/context/NotificationContext';
+
+export default function CreateChatroomPage() {
+  const { chatroomId } = useParams();
+  const navigate = useNavigate();
+  const { goBack } = useNavigation();
+  const { warn, notify } = useNotificationContext();
+
+  const getChatRoomById = useCallback(
+    () => chatApi.chatControllerGetChatroomById({ id: chatroomId! }),
+    [chatroomId],
+  );
+  const { data: chatroom, isLoading } = useData(getChatRoomById);
+  const [isPublic, setIsPublic] = useState(false);
+
+  const initialFormValues: UpdateChatroomDto = {
+    name: isLoading ? '' : chatroom!.name,
+    oldPassword: '',
+    password: '',
+    confirmationPassword: '',
+    isPublic: isPublic,
+  };
+  const [formValues, setFormValues] =
+    useState<UpdateChatroomDto>(initialFormValues);
+
+  useEffect(() => {
+    if (chatroom) {
+      setIsPublic(chatroom.isPublic);
+    }
+  }, [chatroom]);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setFormValues((previousValues: any) => {
+      return { ...previousValues, [name]: value };
+    });
+  };
+
+  function hasValidFormValues() {
+    if (formValues.password !== formValues.confirmationPassword) {
+      warn('Passwords are different');
+      return false;
+    }
+    return true;
+  }
+
+  async function updateChatroomDetails() {
+    if (!hasValidFormValues()) {
+      return;
+    }
+    try {
+      await chatApi.chatControllerUpdateChatroom({
+        chatroomId: chatroomId!,
+        updateChatroomDto: {
+          name: formValues.name ? formValues.name : initialFormValues.name,
+          oldPassword: formValues.oldPassword || null,
+          password: formValues.password || null,
+          confirmationPassword: formValues.confirmationPassword || null,
```

Comment:
thought: There are four cases here:

1. The user updates a public chatroom and doesn't change the password (from public to public -> `chatroom.public && isPublic`). We don't need to send the old (current) password + password + confirmation to the backend.
2. The user adds a password to a public chatroom (from public to protected -> `chatroom.public && isPublic ===false`). The old (current) password is `null`, and we need to send the password and confirmation password to the backend.
3. The user wants to remove the password of the chatroom (from protected to public -> `chatroom.public === false && isPublic`). We need to send the old password for verification, and the new password and confirmation passwords are null.
4. The user updates a protected chatroom without changing it's type (from protected to protected -> `chatroom.public === false && isPublic === false`). All the passwords are either set (in case they want to change the password) or `undefined` (they don't want to change the password).

Or in code, something like:

<details>

```typescript
      let updateChatroomDto: UpdateChatroomDto;
      // Update public chatroom
      if (chatroom.isPublic && isPublic) {
        updateChatroomDto = {
          name: formValues.name || undefined,
        };
        // Update from public to protected
      } else if (chatroom.isPublic && !isPublic) {
        updateChatroomDto = {
          name: formValues.name || undefined,
          oldPassword: null,
          password: formValues.password,
          confirmationPassword: formValues.confirmationPassword,
        };
        // Update from protected to public
      } else if (!chatroom.isPublic && isPublic) {
        updateChatroomDto = {
          name: formValues.name || undefined,
          oldPassword: formValues.oldPassword,
          password: null,
          confirmationPassword: null,
        };
        // Update from protected to protected
      } else {
        updateChatroomDto = {
          name: formValues.name || undefined,
          oldPassword: formValues.oldPassword || undefined,
          password: formValues.password || undefined,
          confirmationPassword: formValues.confirmationPassword || undefined,
        };
      }
```

</details>

Reply:
Corrected with your suggestion